### DISCUSSION TITLE
Make PR template field mandatory

### DISCUSSION THREAD

**Discussion by shailesh-vaidya** [community/community#23848] [New to GitHub]
<p>Hello Team,<br>
Currently, we can add a PR template and make it mandatory however users can keep it empty or keep fields as default without editing it.  Can we make PR template fields mandatory by any method?<br>
Maybe one way to create GitHub action is to query PR and check that required fields are selected.</p>
<p>Regards,<br>
Shailesh</p>

**Comment by mpboom** [comment, marked as answer]
<aside class="quote no-group" data-username="shailesh-vaidya" data-post="1" data-topic="203194">
<div class="title">
<div class="quote-controls"></div>
 shailesh-vaidya:</div>
<blockquote>
<p>Maybe one way to create GitHub action is to query PR and check that required fields are selected.</p>
</blockquote>
</aside>
<p>I think that’s the way to go for now - externally check if the submitted PR complies to your guidelines.</p>
<p>PR <em>templates</em> are just that - templates. You can just clear the text field and type anything you want; enforcement is currently not possible.</p>

**Reply by christian-marquez** [reply]
Is this still not possible? The solution i created for this is to read the PR message via bash script to check for the fields needed then fail a github action to flag it failed.

**Reply by madhurikulkarni** [reply]
> Is this still not possible? The solution i created for this is to read the PR message via bash script to check for the fields needed then fail a github action to flag it failed.

Could you share an example if possible ? trying to see if the test data fields fields are not left empty

**Reply by mpboom** [reply]
It has been four years, and I think now there are more options. From the docs (https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates) here:

> You can specify different input types and set inputs as required to help contributors open actionable issues in your repository. For more information, see [Configuring issue templates for your repository](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) and [Syntax for issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms).

So I'd look into that "new" issue templating flow :)
